vm/actor/src/builtin/chaos/state.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use encoding::Cbor;
+use serde::de::{self, Deserializer};
+use serde::ser::{self, Serializer};
+use serde::{Deserialize, Serialize};
+
+#[derive(Default)]
+pub struct UnmarshallableCBOR {}
+
+impl Serialize for UnmarshallableCBOR {
+    fn serialize<S>(&self, _serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        Err(ser::Error::custom(
+            "Automatic fail when serializing UnmarshallableCBOR",
+        ))
+    }
+}
+
+impl<'de> Deserialize<'de> for UnmarshallableCBOR {
+    fn deserialize<D>(_deserializer: D) -> Result<Self, <D as Deserializer<'de>>::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        Err(de::Error::custom(
+            "Automatic fail when deserializing UnmarshallableCBOR",
+        ))
+    }
+}
+
+impl Cbor for UnmarshallableCBOR {}
+
```

Comment:
Why not just use a shared structure (same as from puppet actor?)

Also, just a slight pref thing, but all of these test actors could be placed in their own folder/module as to not be grouped with the actual builtin actors (right now they are in the builtin directory but puppet and chaos could be in utils or just in a seperate module). This part can be ignored now though

Reply:
> Why not just use a shared structure (same as from puppet actor?)
> 
> Also, just a slight pref thing, but all of these test actors could be placed in their own folder/module as to not be grouped with the actual builtin actors (right now they are in the builtin directory but puppet and chaos could be in utils or just in a seperate module). This part can be ignored now though

I think it makes sense to move it now instead of putting it off for later. If I am moving it tho, should the constants like CHAOS_ACTOR_ADDR still be in vm/actor/src/builtin or should they be moved as well?

Reply:
I think it makes sense to keep the addr and code cids together, no need to move those out as well, unless you want to put them in the same module as each actor themselves (up to you!)